sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -220,6 +222,404 @@ case class EndsWith(left: Expression, right: Expression)
+
+      UTF8String.fromString(sb.toString)
+    }
+  }
+
+  override def toString: String = s"printf($format, $args)"
+}
+
+/**
+ * Returns the string which repeat the given string value n times.
+ */
+case class StringRepeat(str: Expression, times: Expression)
+  extends Expression with ExpectsInputTypes {
+
+  override def children: Seq[Expression] = str :: times :: Nil
+  override def foldable: Boolean = str.foldable && times.foldable
+  override def nullable: Boolean = str.nullable || times.nullable
+  override def dataType: DataType = StringType
+  override def expectedChildTypes: Seq[DataType] = Seq(StringType, IntegerType)
+
+  override def eval(input: InternalRow): Any = {
+    val s = str.eval(input)
+    if (s == null) {
+      null
+    } else {
+      val t = times.eval(input)
+      if (t == null) {
+        null
+      } else {
+        UTF8String.fromString(s.asInstanceOf[UTF8String].toString() * t.asInstanceOf[Integer])
+      }
+    }
+  }
+
+  override def toString: String = s"repeat($str, $times)"
+}
+
+/**
+ * Returns the reversed given string.
+ */
+case class StringReverse(child: Expression) extends UnaryExpression with String2StringExpression {
+  override def convert(v: UTF8String): UTF8String = v.reverse()
+
+  override def toString: String = s"reverse($child)"
+
+  override def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
+    defineCodeGen(ctx, ev, c => s"($c).reverse()")
+  }
+}
+
+/**
+ * Returns a n spaces string.
+ */
+case class StringSpace(child: Expression) extends UnaryExpression with ExpectsInputTypes {
+
+  override def dataType: DataType = StringType
+  override def expectedChildTypes: Seq[DataType] = Seq(IntegerType)
+
+  override def eval(input: InternalRow): Any = {
+    val s = child.eval(input)
+    if (s == null) {
+      null
+    } else {
+      val length = s.asInstanceOf[Integer]
+
+      val spaces = new Array[Byte](if (length < 0) 0 else length)
+      java.util.Arrays.fill(spaces, ' '.asInstanceOf[Byte])
+      UTF8String.fromBytes(spaces)
+    }
+  }
+
+  override def toString: String = s"space($child)"
+}
+
+/**
+ * Splits str around pat (pattern is a regular expression).
+ */
+case class StringSplit(str: Expression, pattern: Expression)
+  extends Expression with ExpectsInputTypes {
+
+  override def foldable: Boolean = str.foldable && pattern.foldable
+  override def nullable: Boolean = str.nullable || pattern.nullable
+  override def dataType: DataType = ArrayType(StringType)
+  override def expectedChildTypes: Seq[DataType] = StringType :: StringType :: Nil
+  override def children: Seq[Expression] = str :: pattern :: Nil
+
+  override def eval(input: InternalRow): Any = {
+    val v = str.eval(input)
+    if (v == null) {
+      null
+    } else {
+      val p = pattern.eval(input)
+      if (p == null) {
+        null
+      } else {
+        val splits = split(v.asInstanceOf[UTF8String], p.asInstanceOf[UTF8String], -1)
```

Comment:
I guess you can express this with Scala more concise:
```
val splits = v.asInstanceOf[UTF8String].toString.split(pattern.asInstanceOf[UTF8String].toString, -1)
splits.toSeq.map(UTF8String.fromString)
```

I tested it for the string `"a,a,a,a,a,a,,a,aa,a,a,,a,a,aa"` and the pattern `,`, it's even a little bit faster (1,000,000 times 1851ms vs 1687ms).